Agents/GFAAgent/src/main/java/uk/ac/cam/cares/jps/agent/gfaagent/GFACalculation.java

```diff
@@ -0,0 +1,54 @@
+package uk.ac.cam.cares.jps.agent.gfaagent;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.apache.jena.vocabulary.AS;
+import org.json.JSONArray;
+
+import uk.ac.cam.cares.jps.base.query.RemoteRDBStoreClient;
+import uk.ac.cam.cares.jps.base.exception.JPSRuntimeException;
+public class GFACalculation {
+
+    private final String dbUrl;
+    private final String user;
+    private final String password;
+
+    private RemoteRDBStoreClient postgisClient;
+
+    public GFACalculation (String postgisDb, String postgisUser, String postgisPassword){
+        this.dbUrl = postgisDb;
+        this.user = postgisUser;
+        this.password = postgisPassword;
+
+        this.postgisClient = new RemoteRDBStoreClient(dbUrl, user, password);
+    }
+    
+    /******************************************** */
+    /* Calculate GFA and store in citydb.cityobject_genericattrib */
+    /**********************************************/ 
+
+    public void calculationGFA(){
+        try (Connection srcConn = postgisClient.getConnection()) {
+            try (Statement stmt = srcConn.createStatement()) {
+                stmt.executeUpdate(gfaSQLInsert);
+            }
+        }catch (SQLException e) {
+            throw new JPSRuntimeException("Error connecting to source database: " + e);
+        }  
+    }
+
+    private static final String gfaSQLInsert = "INSERT INTO cityobject_genericattrib (attrname, realval, cityobject_id)\n" +
+                                                "SELECT DISTINCT ON (attrname, cityobject_id) *\n" +
+                                                    "FROM  (\n" +
+                                                    "SELECT 'GFA',\n" +
+                                                    "(CASE WHEN building.storeys_above_ground IS NOT NULL THEN (public.ST_Area(surface_geometry.geometry)*building.storeys_above_ground)\n" +
+                                                        "ELSE (building.measured_height/3.2*public.ST_Area(surface_geometry.geometry)) END) AS realval, building.id\n" +
```

Comment:
I am assuming that you made the assumption that building heights are 3.2 if no floor data is available? If so, please add this assumption to the README. Also if you have time, please create a request parameter that allows users to change this building height value so that it is generalisable.